RELEASE.md

```diff
@@ -0,0 +1,628 @@
+# Release Documentation
+
+## Quick Start
+
+### Release Turborepo CLI
+
+1. Create a release by triggering the [Turborepo Release][1] workflow
+
+   - Specify the semver increment using the SemVer Increment field (start with `prerelease`)
+   - Check the "Dry Run" box to run the full release workflow without publishing any packages.
+
+2. A PR is automatically opened to merge the release branch created in step 1 back into `main`
+
+   - ⚠️ Merge this in! You don't need to wait for tests to pass. It's important to merge this branch soon after the
+     publish is successful
+
+### Release `@turbo/repository`
+
+1. Run [`bump-version.sh`][4] to update the versions of the packages. Merge in the changes to `main`.
+
+2. Create a release by triggering the [Turborepo Library Release][5] workflow.
+   - Check the "Dry Run" box to run the full release workflow without publishing any packages.
+
+### Notes
+
+- GitHub Release Notes are published automatically using the config from [`turborepo-release.yml`][2],
+  triggered by the [`turbo-orchestrator`][3] bot.
+
+---
+
+## Turborepo CLI Release Process - In-Depth Guide
+
+This section provides comprehensive documentation on how the Turborepo CLI is released, including the architecture, workflows, and detailed step-by-step processes.
+
+### Table of Contents
+
+1. [Architecture Overview](#architecture-overview)
+2. [Prerequisites](#prerequisites)
+3. [Version Management](#version-management)
+4. [Release Workflow Stages](#release-workflow-stages)
+5. [Packages Released](#packages-released)
+6. [Platform-Specific Binaries](#platform-specific-binaries)
+7. [Troubleshooting](#troubleshooting)
+8. [Technical Reference](#technical-reference)
+
+---
+
+### Architecture Overview
+
+The Turborepo release process is a multi-stage pipeline that:
+
+1. **Manages versions centrally** via `version.txt` at the repository root
+2. **Builds Rust binaries** for 6 different platforms (macOS, Linux, Windows on x64 and ARM64)
+3. **Packages native binaries** as separate npm packages (e.g., `turbo-darwin-64`, `turbo-linux-arm64`)
+4. **Publishes JavaScript packages** (main `turbo` package, `create-turbo`, codemods, ESLint plugins, etc.)
+5. **Creates a staging branch** with version bumps and automatically opens a PR to merge back to `main`
+
+The entire process is orchestrated through a GitHub Actions workflow located at `.github/workflows/turborepo-release.yml`.
+
+---
+
+### Prerequisites
+
+Before triggering a release, ensure:
+
+1. **npm Access**: You have publishing rights to the `vercel` npm organization
+2. **GitHub Permissions**: You can trigger GitHub Actions workflows on the `vercel/turborepo` repository
+3. **Clean Working Directory**: No unpushed commits on `main` (the workflow will verify this)
```

Comment:
Also doesn't make sense here. Maybe if you were releasing from local, but we don't do that. 
```suggestion
```